### PR TITLE
fix compilation with gcc

### PR DESCRIPTION
**Pull request by KraemerDEM**
I'm currently working on making couchbaselite compatible with gcc.
This is the first part, there are other changes needed (in couchbase-lite-core) .

### CODE REVIEW

**Comment by avsej** [comment]
Will you also make Fleece gcc-friendly? https://github.com/couchbaselabs/fleece/pull/25

**Comment by KraemerDEM** [comment]
Well, as your pull request already has the necessary changes, there is no need to open another one.
Only one of the devs needs to pull it...

File: src/blip/BLIPConnection.cc [hunk 1]
```diff
@@ -626,7 +626,7 @@ namespace litecore { namespace blip {
 
     void Connection::setWebSocket(WebSocket *webSocket, const fleeceapi::AllocedDict &options) {
         _compressionLevel = kDefaultCompressionLevel;
-        auto levelP = options[kCompressionLevelOption];
+        auto levelP = options.get(kCompressionLevelOption);
```
Comment (snej) [inline src/blip/BLIPConnection.cc:629]: Why doesn't `operator[]` work here?

Comment (KraemerDEM) [reply src/blip/BLIPConnection.cc:629]: gcc complains about ambiguity of the [] operator, I don't know why it isn't able to take the one with parameter const char*.
The same happens in https://github.com/couchbase/couchbase-lite-core/pull/494 .

**Review by snej — APPROVED** [review]

**Pull request merged**